### DISCUSSION TITLE
with 'uvicorn==0.21.0' modifications made to the 'request.state' persist between requests

### DISCUSSION THREAD

**Discussion by yromanyshyn** [Kludex/uvicorn#1901] [Potential Issue] [answered]
support of the `lifespan state` and the notion of the `app_state` was added to the latest release of the uvicorn, as result, now changes made to the `request.state` persist between requests (actually in a very weird way/), IMO, it is confusing and unexpected behavior, you do not expect that changes made to the request state will outlive the request/

**Code snippet:**

```python
from starlette.applications import Starlette
from starlette.responses import PlainTextResponse
from starlette.routing import Route

def handler(request):
    if not hasattr(request.state, "foo"):
        request.state.foo = 10
    else:
        request.state.foo += 1
    return PlainTextResponse(f"Foo: {request.state.foo}")

app = Starlette(debug=True, routes=[Route('/', handler),])
```

**Demonstration:**
(_please, look the video to the end_)

https://user-images.githubusercontent.com/71635444/225558733-8ae7f402-168f-4cd1-b491-01f4a21329ed.mp4

**Installed packages and python version:**

```
>> pip freeze
anyio==3.6.2
click==8.1.3
h11==0.14.0
idna==3.4
sniffio==1.3.0
starlette==0.26.1
typing-extensions==4.5.0
uvicorn==0.21.0

>> python --version
Python 3.8.10
```

**Comment by Kludex** [comment, marked as answer]
It's a bug on uvicorn, not on Starlette.

It looks like the state is persisting per connection, not per request.

cc @adriangb (I can fix it later, just letting you know)
